pkg/client/googlegroup_github/mapper.go

```diff
@@ -19,15 +19,85 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"slices"
 	"strings"
 
 	"google.golang.org/protobuf/encoding/prototext"
 
 	"github.com/abcxyz/pkg/logging"
 	"github.com/abcxyz/team-link/apis/v1alpha3"
+	"github.com/abcxyz/team-link/pkg/github"
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
+// GroupMapper implements groupsync.OneToManyGroupMapper
+// For now it was be used as Mapper between GoogleGroup
+// and GitHubTeam.
+// Both GoogleGroupMapper and GitHubMapper have similar
+// variables and functions, this way we don't need to repeatly
+// write similar function for both on them.
+type GroupMapper struct {
+	mappings map[string][]string
+}
+
+func (m *GroupMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
+	res := make([]string, 0, len(m.mappings))
+	for key := range m.mappings {
+		res = append(res, key)
+	}
+	slices.Sort(res)
+	return res, nil
+}
+
+func (m *GroupMapper) ContainsGroupID(ctx context.Context, key string) (bool, error) {
+	_, ok := m.mappings[key]
+	return ok, nil
+}
+
+func (m *GroupMapper) MappedGroupIDs(ctx context.Context, key string) ([]string, error) {
+	x, ok := m.mappings[key]
+	if !ok {
+		return nil, fmt.Errorf("no mapping found for group ID: %s", key)
+	}
+	// Make deep copy so the caller's operation on return won't change
+	// the value of this given map.
+	ret := make([]string, len(x))
+	copy(ret, x)
+	return ret, nil
+}
+
+type BiDirectionalGroupMapper struct {
+	SourceMapper *GroupMapper
+	TargetMapper *GroupMapper
+}
+
+// NewBidirectionalGoogleGroupGitHubMapper creates a GoogleGroupToGitHubMapper
+// and a GitHubToGoogleGroupMapper using the provided groupMapping file.
+// Returns is (GoogleGroupToGitHubMapper, GitHubToGoogleGroupMapper, error).
+//
+// TODO: refactor this into client/googlegroup_github/mapper.go later.
+func NewBidirectionaGroupMapper(groupMappingFile string) (*BiDirectionalGroupMapper, error) {
+	b, err := os.ReadFile(groupMappingFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read mapping file: %w", err)
+	}
+	tm := &v1alpha3.GoogleGroupToGitHubTeamMappings{}
```

Comment:
```suggestion
	var tm v1alpha3.GoogleGroupToGitHubTeamMappings
```

Then `&tm` throughout.